src/veranda/raster/mosaic/netcdf.py

```diff
@@ -0,0 +1,595 @@
+
+        """
+
+        super().__init__(mosaic, file_register=file_register, data=data, stack_dimension=stack_dimension,
+                         stack_coords=stack_coords, dirpath=dirpath, fn_pattern=fn_pattern, fn_formatter=fn_formatter)
+
+    @classmethod
+    def from_data(self, data, filepath, mosaic=None, **kwargs):
+        """
+        Creates `NetCdfWriter` instance from an xarray.Dataset instance and a target file path, i.e. this function
+        should help to write/export the whole image stack to one file.
+
+        Parameters
+        ----------
+        data : xr.Dataset
+            Dataset to write to disk.
+        filepath : str
+            Full system path to NetCDF file to write to.
+        mosaic : geospade.raster.MosaicGeometry, optional
+            Mosaic representing the spatial allocation of the given file. The tiles of the mosaic have to match the
+            ID's/names of the 'tile_id' column. If it is None, a one-tile mosaic will be created from the given
+            mosaic.
+        kwargs : dict, optional
+            Key-word arguments for initialising the `NetCdfWriter` class.
+
+        Returns
+        -------
+        NetCdfWriter
+
+        """
+        file_register_dict = dict()
+        file_register_dict['tile_id'] = ['0']
+        file_register_dict['filepath'] = [filepath]
+        file_register = pd.DataFrame(file_register_dict)
+        return super().from_xarray(data, file_register, mosaic=mosaic, **kwargs)
+
+    def write(self, data, encoder=None, encoder_kwargs=None, overwrite=False, unlimited_dims=None, **kwargs):
+        """
+        Writes a certain chunk of NetCDF data to disk.
+
+        Parameters
+        ----------
+        data : xr.Dataset
+            Data chunk to be written to disk or being appended to existing mosaic.
+        encoder : callable, optional
+            Function allowing to encode a xarray dataset before writing it to disk.
+        encoder_kwargs : dict, optional
+            Keyword arguments for the encoder.
+        overwrite : bool, optional
+            True if the NetCDF file(s) should be overwritten, False if not (default).
+        unlimited_dims : list of str or str, optional
+            List of dimension names specifying the dimensions to be stored as unlimited.
+        kwargs : dict, optional
+            Key-word arguments for creating a `NetCdf4File` instance.
+
+        """
+        data_geom = self.raster_geom_from_data(data, sref=self.mosaic.sref)
+        unlimited_dims = to_list(unlimited_dims)
+        data_variables = list(data.data_vars)
+        all_dims = list(data.dims)
+        space_dims = all_dims[-2:]
+        stack_dim_names = all_dims[:-2]
+        stack_dims = dict()
+        for stack_dim_name in stack_dim_names:
+            if stack_dim_name in unlimited_dims:
+                stack_dims[stack_dim_name] = None
+            else:
+                stack_dims[stack_dim_name] = len(data[stack_dim_name])
```

Comment:
extract: `stack_dims = {...`

also looks to me as if you could reuse GeoTiff logic here

Reply:
Done.